openedx/features/discussion_provider_piazza/views.py

```diff
@@ -0,0 +1,170 @@
+"""
+Piazza Discussion Provider views.
+"""
+from collections import namedtuple
+from typing import Dict
+from urllib.parse import quote
+
+from django.contrib.auth.models import User
+from django.contrib.sites.shortcuts import get_current_site
+from django.http import HttpRequest
+from lti_consumer.lti_1p1.contrib.django import lti_embed
+from opaque_keys.edx.keys import CourseKey
+from web_fragments.fragment import Fragment
+
+from lms.djangoapps.courseware.access import get_user_role
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.discussions.api.config import get_course_discussion_config
+from openedx.core.djangoapps.discussions.api.data import CourseDiscussionConfigData
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+from openedx.core.djangolib.markup import HTML
+from student.models import anonymous_id_for_user
+
+
+LtiCredentials = namedtuple('LtiCredentials', ('oauth_key', 'oauth_secret'))
+
+
+class LtiCourseLaunchMixin:
```

Comment:
Once we create a yellodig plugin we'll have a better idea of what all LTI code can be kept in common, and come up with an generic interface that multiple plugins can reuse. 

Essentially, what we need is code that helps in building a LTI launch that automatically derives as many correct settings as possible from the current context. 

Here it will only support the course context, but for LTI embeds inside a unit or elsewhere this should be able to support other contexts as well. This can potentially be done by overriding attributes or methods.